packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.tsx

```diff
@@ -201,8 +197,10 @@ function DocSearch({
       [closeModal],
     );
 
-  const transformSearchClient = useCallback(
-    (searchClient: SearchClient) => {
+  const transformSearchClient = useCallback<
+    NonNullable<DocSearchModalProps['transformSearchClient']>
+  >(
+    (searchClient) => {
```

Comment:
```suggestion
    (searchClient: DocSearchTransformClient) => {
```

Couldn't we use the DocSearchTransformClient type here?